rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -157,15 +148,16 @@ public synchronized void addNode(Node node) {
                     toPing.add(nodesOnHost.get(h).iterator().next());
                 }
             }
+            HashMap<InetAddress, Long> hostsTopology = null;
 
-            HashMap<InetAddress, Long> hostsTopology = pingNode(node, toPing);
+            if (PAResourceManagerProperties.RM_TOPOLOGY_DISTANCE_ENABLED.isSet() && PAResourceManagerProperties.RM_TOPOLOGY_DISTANCE_ENABLED.getValueAsBoolean()) {
```

Comment:
This condition could be put in a new method to factorize code and improve readability.